sdk-lib/src/main/java/cash/z/ecc/android/sdk/SdkSynchronizer.kt

```diff
@@ -416,7 +420,7 @@ class SdkSynchronizer private constructor(
             coroutineScope.launch {
                 Twig.info { "Stopping synchronizer $synchronizerKey…" }
                 processor.stop()
-                fetchExchangeChangeUsd.dispose()
+                torClient.dispose()
```

Comment:
The other reason for having each `TorClient` holder hold its own, is that if there is any ongoing background task using this `torClient` after this disposal call, it would throw an `IllegalStateException` due to the `checkNotNull` in `TorClient` failing.

If instead each `TorClient` holder holds its own (all obtained from the same root `TorClient.new()` call), then the background tasks can exit cleanly because the backing Rust Tor client only shuts down when the last `TorClient.dispose()` is called.